libs/window-ref/src/lib/window-ref.service.ts

```diff
@@ -19,10 +21,39 @@ export class WindowRefService {
     return this.window.document;
   }
 
+  public getHash() {
+    return this.window.location.hash;
+  }
+
   public getIndexedDB() {
     return this.window.indexedDB;
   }
 
+  public getPlopdownFromHash(): Observable<string> {
+    const plopdowns = new Observable<string>(subscriber => {
+      this.emitPlopdownFromHash(subscriber);
+      this.window.addEventListener('hashchange', function() {
+        this.emitPlopdownFromHash(subscriber);
+      }.bind(this), false);
+    });
```

Comment:
Once the above change is made, you can move the `emitPlopdownFromHash` logic into the constructor as a `map` operator, set to a variable:
```ts
this.plopdownFromHash$ = this.hashChange$.pipe(
    map(hash => this.parsePlopdownFromHash(hash)),
    filter(video => video != null)
)
```

then just return that observable here.

Reply:
- how would you like the initial case handled (currently called right before the event handler is set up)?

something like?
```javascript
this.hashChange$.pipe(
    startWith(this.getHash()),
    map(hash => this.parsePlopdownFromHash(hash)),
    filter(video => video != null),
)
```

- this change looks like it would use the same reference (`this.plopdownFromHash$`) for each call to `getPlopdownFromHash`. Would that cause multiple calls to stomp on eachother? Or at least risk odd behavior?

Reply:
sorry for the late reply! you can use `startWith`, or initialize the `hashChange$` subject as a `BehaviorSubject`, either works.

Reply:
[`BehaviorSubject`](https://rxjs.dev/guide/subject#behaviorsubject) looks right